[stdlib] Conditional gardening [NFC]

moiseev opened the pull request.

Commit:
[stdlib] Conditional gardening

moiseev commented:
@swift-ci Please smoke test

DougGregor reviewed (commented):
There are three different indentations for the `where` in a where clause in this PR.

moiseev commented:
@DougGregor yes.

airspeedswift commented on stdlib/public/core/Indices.swift:
```diff
@@ -130,5 +130,12 @@ extension Collection where Indices == DefaultIndices<Self> {
+@available(*, deprecated, renamed: "DefaultIndices")
+public typealias DefaultBidirectionalIndices<
+  T : BidirectionalCollection
+> = DefaultIndices<T>
```
I feel like you're just 80-col-trolling me now...

moiseev replied:
Reverted. It doesn't make much sense here.

airspeedswift reviewed (approved):
LGTM

moiseev commented:
@swift-ci Please smoke test

moiseev commented:
@swift-ci Please smoke test and merge

moiseev commented:
@swift-ci Please smoke test and merge

moiseev commented:
For the record. The failure is:

```
--
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:193:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:193:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:414:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:414:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:599:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:599:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:832:23: error: unexpected warning produced: 'DefaultRandomAccessIndices' is deprecated: renamed to 'DefaultIndices'
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRangeReplaceableRandomAccessButt>
                      ^
/Users/buildnode/jenkins/workspace/swift-PR-osx-smoke-test/branch-master/buildbot_incremental/swift-macosx-x86_64/test-macosx-x86_64/stdlib/Output/collection-combinatorics.swift.gyb.tmp/collection-combinatorics.swift:832:23: error: unexpected note produced: use 'DefaultIndices' instead
  typealias Indices = DefaultRandomAccessIndices<CustomMutableRangeReplaceableRandomAccessButt>
                      ^

--
```

DougGregor reviewed (approved):
LGTM

moiseev commented:
@swift-ci Please test and merge

moiseev commented:
@swift-ci Please test

airspeedswift commented:
@swift-ci please test macOS platform
